source/python/idetect/interpreter.py

```diff
@@ -71,6 +71,29 @@ def get_absolute_date(relative_date_string, publication_date=None):
         return None
 
 
+def cleanup(text):
+    """
+    Cleanup text based on commonly encountered errors.
+    param: text     A string
+    return: A cleaned string
+    """
+    text = re.sub(r'([a-zA-Z0-9])(IMPACT)', r'\1. \2', text)
+    text = re.sub(r'([a-zA-Z0-9])(RESPONSE)', r'\1. \2', text)
+    text = re.sub(r'(IMPACT)([a-zA-Z0-9])', r'\1. \2', text)
+    text = re.sub(r'(RESPONSE)([a-zA-Z0-9])', r'\1. \2', text)
+    text = re.sub(r'([a-zA-Z])(\d)', r'\1. \2', text)
+    text = re.sub(r'(\d)\s(\d)', r'\1\2', text)
+    text = text.replace('\r', ' ')
+    text = text.replace('  ', ' ')
+    text = text.replace('\n', ' ')
+    text = text.replace("peole", "people")
+    output = ''
```

Comment:
I like using `''.join()` to make strings from parts. eg. `''.join([c for c in text if c in string.printable])`

Reply:
Thanks for all the pointers on this function.
The first rules regarding the words RESPONSE / IMPACT are because, based on the initial urls we were given, there seemed to be a lot of reports from a wire service that summarizes events and seems to have those as  keywords but often with missing spaces.

For the moment I have simplified to:
```
def cleanup(text):
    """
    Cleanup text based on commonly encountered errors.
    param: text     A string
    return: A cleaned string
    """
    text = re.sub(r'([a-zA-Z0-9])(IMPACT|RESPONSE)', r'\1. \2', text)
    text = re.sub(r'(IMPACT|RESPONSE)([a-zA-Z0-9])', r'\1. \2', text)
    text = re.sub(r'([a-zA-Z])(\d)', r'\1. \2', text) 
    text = re.sub(r'(\d)\s(\d)', r'\1\2', text)
    text = text.replace('\s+', ' ')
    text = text.replace("peole", "people")
    output = ''.join([c for c in text if c in string.printable])
    return output
```

However, there are definitely opportunities for future improvement here including:
- Using a more general library / approach to correcting spelling mistakes
- Generalizing rules for missing spaces in a way that leads to best outcomes for NLP